src/containers/HomeContainer/Home.js

```diff
@@ -128,6 +131,52 @@ const Home = ({ starred, navigation }) => {
     )
   }
 
+  /**
+   * Render department's course offerings in a dropdown picker based on department id.
+   */
+  const renderDepartmentsDropDown = () => {
+    const [departments, setAllDepartments] = useState([])
+    useEffect(() => {
+      const fetchDepartments = async () => {
+        const allDept = await getUniversityDepartments(universityId)
+        setAllDepartments(allDept)
+      }
+      fetchDepartments()
```

Comment:
Could you use the loadingWrap on fetchDepartments?

Reply:
It actually breaks/complains when I try to wrap the fetch in loadingWrap() and it *seems* to conflict with the loading wrap on the courses rendering.
`f is not a function. (In 'f()', 'f' is an instance of a Promise)`. 

Also, it still displays the loading sign when waiting for the courses to render. Functionality still makes sense even if I don't include it for fetching departments and universities (because rendering courses is dependent on the queries for universities and downloads to finish).

I can keep debugging this, but tbh I'm not to sure where I'm creating a new Promise

Reply:
When you're calling it, are you doing `loadingWrap(fetchDepartments)` or `loadingWrap(fetchDepartments())`? You have to make sure you **don't** have the parenthesis after the function name

Reply:
Yeah, was calling loadingWrap(fetchDepartments()).  

However, it's still not functioning as expected. When we use the loading wrap with fetchDepartments, it will start circling then prematurely end.

Reply:
Sounds like you're missing loadingWrap in the deparment/university selection, but dealing with one finishing earlier than the other is a big problem. II pushed a fix that allows us to deal with multiple loadingWraps in a component